earth2studio/models/px/ace2.py

```diff
@@ -0,0 +1,641 @@
+
+try:
+    # Optional dependency: FME
+    from fme.ace.data_loading.batch_data import BatchData, PrognosticState
+    from fme.ace.stepper.single_module import load_stepper
+except ImportError:
+    OptionalDependencyFailure("ace2")
+    BatchData = Any
+    PrognosticState = Any
+
+
+def _npdatetime64_to_cftime(dt64_array: np.ndarray) -> np.ndarray:
+    """Convert np.datetime64[...] array to cftime.DatetimeProlepticGregorian array (vectorized). Only supports up to seconds precision."""
+
+    if len(dt64_array.shape) > 1:
+        # Flatten the array before applying conversion
+        return_shape = list(dt64_array.shape)
+        dt64_array = dt64_array.reshape(-1)
+    else:
+        return_shape = None
+
+    dt_index = pd.to_datetime(dt64_array)
+
+    years = dt_index.year
+    months = dt_index.month
+    days = dt_index.day
+    hours = dt_index.hour
+    minutes = dt_index.minute
+    seconds = dt_index.second
+
+    result = np.fromiter(
+        (
+            cftime.DatetimeProlepticGregorian(y, m, d, H, M, S)
+            for y, m, d, H, M, S in zip(years, months, days, hours, minutes, seconds)
+        ),
+        dtype=object,
+        count=len(dt64_array),
+    )
+
+    if return_shape is not None:
+        result = result.reshape(return_shape)
+    return result
+
+
+def _cftime_to_npdatetime64(cftime_array: np.ndarray) -> np.ndarray:
+    """Convert cftime.DatetimeProlepticGregorian array to np.datetime64[s] array (vectorized-safe). Only supports up to seconds precision.
+    Out-of-range years become NaT.
+    """
+
+    def _convert_single(t: cftime.DatetimeProlepticGregorian) -> np.datetime64:
+        if not (1678 <= t.year <= 2261):
+            return np.datetime64("NaT")
+        return np.datetime64(
+            f"{t.year:04d}-{t.month:02d}-{t.day:02d}T"
+            f"{t.hour:02d}:{t.minute:02d}:{t.second:02d}"
+        )
+
+    vec_convert = np.vectorize(_convert_single, otypes=["datetime64[s]"])
+    return vec_convert(cftime_array)
+
+
+@check_optional_dependencies()
+class ACE2ERA5(torch.nn.Module, AutoModelMixin, PrognosticMixin):
+    """ACE2-ERA5 prognostic model wrapper.
+
+    ACE2 (Ai2 Climate Emulator v2) is a 450M-parameter autoregressive emulator
+    with 6-hour time steps, 1-degree horizontal resolution, and eight vertical
+    layers that exactly conserves global dry air mass and moisture and can be
+    stepped stably for arbitrarily many steps at about 1500 simulated years
+    per wall-clock day. ACE2-ERA5 was trained on the ERA5 dataset and requires
+    forcing data during rollout (see `forcing_data_source` parameter). This
+    wrapper makes use of the ``fme`` package to run model forward passes.
+
+    Parameters
+    ----------
+    checkpoint_path : str
+        Path to a serialized ACE2-ERA5 stepper checkpoint (e.g., tar file).
+    forcing_data_source : DataSource
+        Data source providing forcing data during rollout. Must provide all
```

Comment:
Perhaps the doc string can mention the use of the AI2 data source as a good go to? What do you think? Will more people just use that?